cli/azd/pkg/tools/azcli/azcli.go

```diff
@@ -59,6 +59,8 @@ type AzCli interface {
 	LoginAcr(ctx context.Context, subscriptionId string, loginServer string) error
 	GetContainerRegistries(ctx context.Context, subscriptionId string) ([]*armcontainerregistry.Registry, error)
 	ListAccounts(ctx context.Context) ([]AzCliSubscriptionInfo, error)
+	GetDefaultAccount(ctx context.Context) (*AzCliSubscriptionInfo, error)
+	GetAccount(ctx context.Context, subscriptionId string) (*AzCliSubscriptionInfo, error)
```

Comment:
I wonder if we should actually move this outside of the `azcli` interface into it's own class given that `azcli` is transitioning to mean "Azure operations" instead of `az cli`. Otherwise, we'll still need to rework this part when we actually remove az CLI.

Reply:
Yes, lets do this as part of the `AzCli` interface refactor.